TM1py/Services/ElementService.py

```diff
@@ -275,3 +276,21 @@ def get_members(element):
 
         get_members(consolidation_tree)
         return members
+
+    def execute_set_mdx(self, mdx, top=10000):
+        '''
+        :param mdx: valid dimension mdx statement
+        :param top: number of records to return
+        :return: dictionary of members, unique names, weights, types, ordinals, and parents
+        '''
+
+        request = f'/api/v1/ExecuteMDXSetExpression?$expand=Tuples($top={top};' \
+                  '$expand=Members($select=Name,Ordinal,Weight;' \
+                  '$expand=Parent($select=Name,UniqueName),' \
+                  'Element($select=Type,Level)))'
+
+        payload = {"MDX": mdx}
+        response = self._rest.POST(request, json.dumps(payload, ensure_ascii=False))
+        raw_dict = response.json()
+        return [tuples['Members'][0] for tuples in raw_dict['Tuples']]
```

Comment:
I think instead of 

``` python
return [tuples['Members'][0] for tuples in raw_dict['Tuples']]
```

it would be better to write
``` python
return [tuples['Members'] for tuples in raw_dict['Tuples']]
```

so that it also works for "Sets of Tuples":

``` python
with TM1Service(address=ADDRESS, port=PORT, user=USER, password=PASSWORD, ssl=SSL) as tm1:

    mdx = "{ ([region].[europe],[month].[jan]) }"

    response = tm1.dimensions.hierarchies.elements.execute_set_mdx(mdx)
    print(response)
```

Reply:
Fixed in the latest commit